src/session/Manager.cc

```diff
@@ -218,19 +218,34 @@ void Manager::Insert(Session* s, bool remove_existing)
 
 void Manager::Drain()
 	{
-	std::vector<const detail::Key*> keys;
-	keys.reserve(session_map.size());
+	// If a random seed was passed in, we're most likely in testing mode and need the
+	// order of the sessions to be consistent. Sort the keys to force that order
+	// every run.
+	if ( zeek::util::detail::have_random_seed() )
```

Comment:
(Just for the record: I just (re-)discovered that we have an explicit "deterministic mode" through `-D`. That one seeds the RNG as well, so it is covered by this check.)